### DISCUSSION TITLE
How to instruct SnowDDL to ignore objects it doesn't maintain?

### DISCUSSION THREAD

**Discussion by minghungcho** [littleK0i/SnowDDL#7] [Ideas]
I have been testing singledb mode for several days, and so far it works as expected. 
However, one issue I encountered is that there is no option to instruct SnowDDL not to drop objects it does not maintain.
Since I also use dbt for transformation, there are tables and views maintained by dbt in the same database.
In addition, I am also using SnowPark to deploy Python UDF and Stored Procedure.

Any suggestions on how to make SnowDDL work with other tools?

**Comment by littleK0i** [comment]
Currently the best way to achieve it is by creating a dedicated schema for objects maintained externally, and by setting option `is_sandbox: true` for that schema: https://docs.snowddl.com/basic/yaml-configs/schema

 Unknown objects in such schemas will not be dropped.

**Reply by psequeirag** [reply]
Given that SnowDDL intentionally ignores objects it doesn't own, we have found a relatively straightforward way to keep it focused only on certain objects in the account (we use this to have separate teams independently maintain their own set of DBs, Roles and Warehouses):
1. First, we create an "admin" role for each team. This role should have CREATE DATABASE, CREATE ROLE and CREATE WAREHOUSE privileges.
2. Also, create one snowddl_admin user for each team, grant this user the corresponding role from above.
3. Ensure each team has a designated, unique and well known PREFIX so that all their object names have that prefix to avoid teams competing for the name of any object (be that DB, role or WH).

And that's it. With the above, each team can independently develop and maintain their own SnowDDL config (and there could also be additional objects not maintained with SnowDDL, too); as long as they consistently run it with their designated snowddl_admin user, they should be able to work on their own subset of objects without messing with objects not owned by them.
